File: app/models/sequence.rb
class Sequence < ActiveRecord::Base
  has_many :sound_patterns
  belongs_to :user

  has_many :sounds, through: :sound_patterns

  def self.prepTemplate
    seq = Sequence.new
    seq.name = 'Name Your Phat Beat!'
    seq.tempo = 120
    seq.steps = 8
    # seq.sounds << Sound.find(1)
    # seq.sounds << Sound.find(2)
    seq

  end

end

File: public/js/models/sequence.js
function Sequence(obj) {
	this.name = obj.name;
	this.sequenceId = obj.sequenceId;
	this.soundPatterns = obj.soundPatterns;
	this.steps = obj.steps;
	this.tempo = obj.tempo;
}

Sequence.prototype = {
}

// function SoundPattern(obj) {
// 	this.sound = obj.sound;
// }

function Sound(obj) {
	this.beats = obj.beats;
	this.url = obj.url;
}

function Beat(obj)  {
	this.play = obj.play;
	this.position = obj.position;
}



